bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerStationHandler.java

```diff
@@ -0,0 +1,129 @@
+package org.openhab.binding.opensprinkler.internal.handler;
+
+import static org.openhab.binding.opensprinkler.internal.OpenSprinklerBindingConstants.STATION_STATE;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.opensprinkler.internal.api.OpenSprinklerApi;
+import org.openhab.binding.opensprinkler.internal.config.OpenSprinklerStationConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OpenSprinklerStationHandler extends OpenSprinklerBaseHandler {
+    private final Logger logger = LoggerFactory.getLogger(OpenSprinklerStationHandler.class);
+
+    private OpenSprinklerStationConfig config;
+
+    public OpenSprinklerStationHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfig().as(OpenSprinklerStationConfig.class);
+
+        super.initialize();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command == RefreshType.REFRESH) {
+                updateChannels();
+            } else {
+                handleStationCommand(config.stationIndex, command);
+            }
+        } catch (Exception exp) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not control the OpenSprinkler.");
+            logger.debug("Error controlling and OpenSprinkler station. Exception received: {}", exp.toString());
+        }
+    }
+
+    /**
+     * Handles control of an OpenSprnkler station based on commanded
+     * received by a channel call.
+     *
+     * @param stationId Int of the station to control. Starts at 0.
+     * @param command Command being issues to the channel.
+     */
+    protected void handleStationCommand(int stationId, Command command) {
+        OpenSprinklerApi api = getApi();
+        if (api == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED,
```

Comment:
see above (`BRIDGE_UNINITIALIZED` should not be set by the thing itself)

Reply:
What would be your suggested handling in this case? Set a COMMUNICATION_ERROR state?

Reply:
Maybe. Or `NONE` and add the reason.

Reply:
Ah, didn't know that NONE exists, sorry for that :) Change it to that one.